packages/data-explorer/src/mappers/mapperUtils.ts

```diff
@@ -0,0 +1,130 @@
+  const fetchOptionsFunction = (search?: string | Array<string>): FilterOptionsPromise => {
+    return fetchFieldOptions((attribute.refEntity), search).then((response: FilterOption[]) => {
+      return response
+    })
+  }
+
+  switch (attribute.fieldType) {
+    case 'CATEGORICAL':
+    case 'CATEGORICAL_MREF':
+      if (attribute.categoricalOptions) {
+        return (): FilterOptionsPromise => Promise.resolve((attribute.categoricalOptions as MetaDataCategoricalOption[]).map(option => {
+          return {
+            value: option.id
+          }
+        }))
+      } else {
+        return fetchOptionsFunction
+      }
+    case 'ONE_TO_MANY':
+    case 'XREF':
+    case 'MREF':
+      return fetchOptionsFunction
+    case 'ENUM':
+      if (attribute.enumOptions) {
+        const enumOptions = attribute.enumOptions.map(option => {
+          return {
+            value: option,
+            text: option
+          }
+        })
+        return (): FilterOptionsPromise => Promise.resolve(enumOptions)
+      } else {
+        return null
+      }
+    case 'BOOL':
+      const boolOptions: FilterOption[] = [
+        { value: true, text: 'Yes' },
+        { value: false, text: 'No' }
+      ]
+      return (): FilterOptionsPromise => Promise.resolve(boolOptions)
+    default:
+      return null
+  }
+}
+
+/**
+ * Uses the idAttribute, labelAttribute, and hrefCollection parameters of the refEntity
+ * to query a data table. Returns a list of {id, value, label} items as a Promise
+ *
+ * @param refEntity The refEntity of the attribute.
+ * @param search An optional search query used to filter the items of the response
+ * @return {Promise} Promise object representing an Array of FieldOption
+ */
+const fetchFieldOptions = (refEntity: MetaDataRefEntity | undefined, search?: string | Array<string>): FilterOptionsPromise => {
+  if (!refEntity) throw new Error('refEntity is undefined, this should not happen')
+
+  const idAttribute = refEntity.idAttribute
+  const labelAttribute = refEntity.labelAttribute ? refEntity.labelAttribute : refEntity.idAttribute
+
+  // map refEntity.hrefCollection v1 URLs to v2 to enable the use of RSQL queries
+  const uri = refEntity.hrefCollection.replace('/v1/', '/v2/')
+  const href = generateUri(uri, idAttribute, labelAttribute, search)
+
+  return api.get(href).then((response: DataApiResponse) => {
+    return response.items.map((item: any): FilterOption => {
+      return {
+        value: item[idAttribute],
+        text: item[labelAttribute]
+      }
+    })
+  })
+}
+
+const generateUri = (
+  uri: string, idAttribute: string, labelAttribute: string, search?: string | Array<string> | null) => {
+  if (search) {
+    if (Array.isArray(search) && search.length > 0) {
+      // Join array into a string
+      const value = search.join(',')
+      // Use =in= query
+      uri = `${uri}?q=${idAttribute}=in=(${value}),${labelAttribute}=in=(${value})`
```

Comment:
Is two different cases: if you're typing in a filter MREF you want to a search query in the lookup attributes. If you're retrieving the labels of a selected range of IDs you want to do an in query on the id attributes.